library/cwm/src/lib/cwm/tree_pager.rb

```diff
@@ -0,0 +1,55 @@
+require "cwm/page"
+require "cwm/pager"
+require "cwm/tree"
+
+module CWM
+  # A {TreeItem} that knows a {Page}, useful for a {TreePager}.
+  # The UI label and `id` are taken from its {Page}: {Page#label},
+  # {Page#widget_id}
+  class PagerTreeItem < TreeItem
+    # @return [Page]
+    attr_reader :page
+
+    # @param page [Page]
+    # @param children [Array<PagerTreeItem>]
+    def initialize(page, icon: nil, open: true, children: [])
+      @page = page
+      super(page.widget_id, page.label,
+            icon: icon, open: open, children: children)
+    end
+
+    # @return [Array<Page>] My page and all pages of descendant items
+    #   (needed to initialize a {Pager}).
+    def pages
+      children.values.flat_map(&:pages).unshift(@page)
+    end
+  end
+
+  # A {Pager} that uses a {Tree} to select the {Page}s
+  class TreePager < Pager
+    # @param items [Array<PagerTreeItem>]
+    def initialize(*items, label:)
+      @tree = Tree.new(items, label: label)
```

Comment:
Ah, I notice it now. Again this design is a bit against CWM phylosophy To have stuff in constructor. So what is my idea how it can looks like? items should be defined in instance subclass same like we have in all items based widgets.
so
```
class MyTree < Tree
  def items
   [
      ...
   ]
  end

  def label
    _("Tree label")
  end
end
```
same way with label.

This way it really confuse me why TreePager have label and now I see that it is label for tree and not for Tree pager ( so different location of pager then I expect ).

Question is then how should look like items when tree is used in TreePager as it need to have different items. Well, answer is that tree already have to know about it, as it is tree widget designed to be used in Pager context. So it is fine for it, if it create TreePagerItems and not simple TreeItems.
That also bring me, that it can be quite easy to add helper to Tree itself to create TreeItem as item and for tree for pager, mention that it have to be pager_item.